### PR TITLE
add np.array_equal support

### LINKED ISSUE #4550: numpy array_equal implementaton
hi folks, could the array_equal function be implemented with @numba.jit ?

it compare shapes are equal, and if all values are equal

https://github.com/numpy/numpy/blob/v1.17.0/numpy/core/numeric.py#L2299-L2339

### PR DESCRIPTION
**Pull request by guilhermeleobas**
Fixes #4550

### CODE REVIEW

File: numba/targets/arraymath.py [hunk 1]
```diff
@@ -3097,6 +3097,20 @@ def diff_impl(a, n=1):
     return diff_impl
 
 
+@overload(np.array_equal)
+def np_array_equal(a, b):
+
+    if not (isinstance(a, types.Array) and isinstance(b, types.Array)):
```
Comment (stuartarchibald) [inline numba/targets/arraymath.py]: Despite having "array" in the function name, this function seems to accept the usual selection of array-like things.
```pycon
In [1]: import numpy as np                                                                                                      

In [2]: np.array_equal((1,2,3),(1,2,3))                                                                                         
Out[2]: True

In [3]: np.array_equal(1,2)                                                                                                     
Out[3]: False

In [4]: np.array_equal(1,1)                                                                                                     
Out[4]: True
```

File: numba/targets/arraymath.py [hunk 2]
```diff
@@ -3097,6 +3097,27 @@ def diff_impl(a, n=1):
     return diff_impl
 
 
+@overload(np.array_equal)
+def np_array_equal(a, b):
+
+    if not (type_can_asarray(a) and type_can_asarray(b)):
+        raise TypingError('Both arguments to "array_equals" must be array-like')
+
+    if isinstance(a, types.Integer) and isinstance(b, types.Integer):
```
Comment (stuartarchibald) [inline numba/targets/arraymath.py]: Why single out integers? Are `types.Scalar` or `types.Boolean` not the same pattern?

Comment (guilhermeleobas) [reply numba/targets/arraymath.py]: Does `types.Scalar` really exists?

Comment (stuartarchibald) [reply numba/targets/arraymath.py]: I probably mean `types.Number` :)

**Comment by esc** [comment]
@guilhermeleobas thanks for the updates, I have queued it again.

File: numba/targets/arraymath.py [hunk 3]
```diff
@@ -3117,6 +3117,28 @@ def diff_impl(a, n=1):
...
 
 
+@overload(np.array_equal)
+def np_array_equal(a, b):
+
+    if not (type_can_asarray(a) and type_can_asarray(b)):
+        raise TypingError('Both arguments to "array_equals" must be array-like')
+
+    if isinstance(a, (types.Integer, types.Boolean, types.Number)) and \
+       isinstance(b, (types.Integer, types.Boolean, types.Number)):
```
Comment (stuartarchibald) [inline numba/targets/arraymath.py]: Recall that `types.Integer` inherits from `types.Number`.
How about:
```python
accepted = (types.Boolean, types.Number)
if isinstance(a, accepted) and isinstance(b, accepted):
```

File: docs/source/reference/numpysupported.rst [hunk 4]
```diff
@@ -299,6 +299,7 @@ Other functions
 
 The following top-level functions are supported:
 
+* :func:`numpy.array_equal`
```
[the remark this answers is not in the document]
Comment (stuartarchibald) [reply docs/source/reference/numpysupported.rst]: This still needs addressing?

File: numba/targets/arraymath.py (hunk 3 above)
Comment (guilhermeleobas) [reply numba/targets/arraymath.py]: Oh, didn't know about that. I've pushed the new version.

**Comment by stuartarchibald** [comment]
waiting on CI after resolving merge conflict.

**Comment by esc** [comment]
For azure only the coverage run timed out. Probably safe to merge.

**Pull request merged**